merge_fastqs.py

```diff
@@ -1,85 +1,43 @@
-# Modified version of https://github.com/SciLifeLab/standalone_scripts/blob/master/merge_and_rename_NGI_fastq_files.py
-# to fit the NGI Uppsala standard
-
+import argparse
 import re
 import os
-import sys
 import shutil
-import argparse
-import collections
 
-def merge_files(input_dir, dest_dir):
+arg_parser = argparse.ArgumentParser(description=""" Merges all fastq-files that match the standard name pattern per sample and read. Looks through the given dir and subdirs.""")
+arg_parser.add_argument("--input_dir", metavar='Input directory', required=True, help="Base directory for the fastq files that should be merged. ")
+arg_parser.add_argument("--output_dir", metavar='Output directory', required=True, help="Path for output of merged files.")
+args = arg_parser.parse_args()
 
-    #Gather all fastq files in inputdir and its subdirs
-    fastq_files=[]
-    for subdir, dirs, files in os.walk(input_dir):
-        for fastq in  files:
-            if fastq.endswith('.fastq.gz'):
-                fastq_files.append(os.path.join(subdir, fastq))
+sample_pattern = re.compile(r"^(.+)_S[0-9]+_L00[1-8]_(R[1-2])_.+\.fastq\.gz$")
 
-    #Match NGI sample number from flowcell
-    sample_pattern=re.compile("^(.+)_S[0-9]+_L00[1-8]_R([1-2])_")
-    #Remove files that already have the right name (i.e have been merged already)
-    matches=[]
-    for fastq_file in fastq_files:
-        try:
-            match=sample_pattern.search(os.path.basename(fastq_file)).group(1)
+def find_fastqs(base_dir, pattern):
+    fastq_dict = {}
+    for root, dirs, files in os.walk(base_dir):
```

Comment:
just a comment here: is `os.walk` guaranteed to always return the files in a predictable order? I assume so but we want to be really sure since the matching of R1 and R2 when setting up the merge below is assuming that the corresponding R1 and R2 files go into the list in the same order..

Reply:
... I think this could be easily resolved by also sorting `fastqs[sample][read]` in the call to `merge_fastqs` below 👇

Reply:
Solved as suggested